### ISSUE TITLE
How can i cancel refreshing event?

### ISSUE THREAD

**Issue by HeyAlex** [liaohuqiu/android-Ultra-Pull-To-Refresh#255]
How can i cancel animation of refreshing and return header in start position?  I need it because if my task return me an error(no connection with in-net) i want to return header in current state and set the last update time. How can i make it?

**Comment by liaohuqiu** [comment]
You can use `refreshComplete()` to cancel / stop.

**Comment by HeyAlex** [comment]
if i use refreshComplete() header will be updated, i mean the counter of last update time will reset. i want to save previous updating time if i've got an error.

**Comment by van048** [comment]
I think you can setup your custom header to do this

**Comment by liaohuqiu** [comment]
I think it will be better to introduce a overload method: `refreshComplete(boolean success)`.

**Comment by van048** [comment]
@liaohuqiu Yeah. It is better in that way. `OnUIRefreshComplete(PtrFrameLayout frame)` should be handled too.

**Comment by HeyAlex** [comment]
As i understand, where is no way for me to extends PtrFrameLayout and override refreshCompleate() ?

**Comment by HeyAlex** [comment]
oh, Merci!
